### PR TITLE
TRUNK-1748:Removed Concept field,Deprecated Concept getters and Sette…

### PR DESCRIPTION
**Pull request by tendomatGitHub**
…rs,Program,ProgramWorkflow,ProgramWorkflowState domain Objects,and Replaced them with  name,description fields respectively.Removed Concept property tags from corresponding hibernate mapping files

https://issues.openmrs.org/browse/TRUNK-1748

Please check this

### CODE REVIEW

**Comment by dkayiwa** [comment]
@tendomatGitHub did you see the above travis failure?

**Comment by tendomatGitHub** [comment]
yes trying to figure out the problem

File: api/src/main/java/org/openmrs/Program.java [hunk 1]
```diff
@@ -122,14 +122,41 @@ public String toString() {
 	// Property Access
 	// ******************
 	
+	/**
+	 * @deprecated As of 2.0.1
+	 * 
+	 * replaced by {@link #getName())}
+	 */
+	 @deprecated
```
Comment (ivange94) [inline api/src/main/java/org/openmrs/Program.java]: this is the culprit. There is no `@deprecated` annotation. This is used only inside javadocs. Like you did above. What you are looking for is `@Deprecated`. Notice capitalization of the D.

Comment (tendomatGitHub) [inline api/src/main/java/org/openmrs/Program.java]: thanks Ivange just refactored that

**Comment by tendomatGitHub** [comment]
https://issues.openmrs.org/browse/TRUNK-1748
@dkayiwa can you please review this

**Comment by tendomatGitHub** [comment]
please review this

**Comment by dkayiwa** [comment]
The formatting in this file is not correct. Did you run mvn clean install before committing?

**Comment by tendomatGitHub** [comment]
i run mvn clean package

**Comment by tendomatGitHub** [comment]
let me do that again

**Comment by tendomatGitHub** [comment]
i run mvn clean install and tests passed...see this http://pastebin.com/RbAaAS2q

**Comment by dkayiwa** [comment]
@tendomatGitHub now commit the formatting changes that have happened after running mvn clean install.

**Comment by tendomatGitHub** [comment]
was also wondering wether am to write changesets to introduce name and descfiption columns in workflow and workflowstate tables

**Comment by dkayiwa** [comment]
@tendomatGitHub are you using an IDE to make these changes? Or are you simply using a text editor like notepad?

File: api/src/main/java/org/openmrs/ProgramWorkflowState.java [hunk 2]
```diff
@@ -58,14 +65,64 @@ public String toString() {
...
+	 * 
+	 */
+	public String getName() {
+		return name;
+	}
+	
+	/**
+	 *  since 2.1.0 
+	 *
+	 *
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/ProgramWorkflowState.java]: Why all these many lines?

File: api/src/main/java/org/openmrs/Program.java [hunk 3]
```diff
@@ -122,14 +126,63 @@ public String toString() {
 	// Property Access
 	// ******************
 	
+	/**
+	 * @deprecated since 2.1.0 replaced with {@link getName() and getDescription()}
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Program.java]: Are you sure getConcept was replaced with getName and getDescription?

**Comment by tendomatGitHub** [comment]
@dkayiwa am using Eclipse though it doesn't recognise openmrs code,no code recommenders for openmrs.

**Comment by tendomatGitHub** [comment]
@dkayiwa let me work on them.

File: api/src/main/java/org/openmrs/Program.java (hunk 3 above)
Comment (tendomatGitHub) [inline api/src/main/java/org/openmrs/Program.java]: @dkayiwa doesn't deprecating mean annotating the code but keeping it for backwards compatibility or waiting for deletion during a new release?
a little puzzled here

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Program.java]: Am asking in reference to your deprecated comment.

**Comment by dkayiwa** [comment]
@tendomatGitHub did you notice the merge conflicts on this pull request? Are you even still working on it?

**Pull request closed without merging**